pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -1196,21 +1196,19 @@ func (r *ReconcilePerconaServerMongoDB) reconcileMongosStatefulset(ctx context.C
 		return errors.Wrapf(err, "create template spec for mongos")
 	}
 
-	if cr.TLSEnabled() {
-		sslAnn, err := r.sslAnnotation(ctx, cr)
-		if err != nil {
-			if err == errTLSNotReady {
-				return nil
-			}
-			return errors.Wrap(err, "failed to get ssl annotations")
-		}
-		if templateSpec.Annotations == nil {
-			templateSpec.Annotations = make(map[string]string)
+	sslAnn, err := r.sslAnnotation(ctx, cr)
+	if err != nil {
+		if err == errTLSNotReady {
```

Comment:
Just make this comparison like `errors.Is(err, errTLSNotReady)` since comparison with errors using equality operators fails on wrapped errors.

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1545/commits/e052291a00ba05897b9f99fb83674d3191dd7d84